### PR TITLE
chore: refactor paginator

### PR DESCRIPTION
**Pull request by chejimmy**
**What this PR does / why we need it**:
- simplify the SiteWise Queries pagination logic:
  - currently, the pagination logic under `src/requestLooper.ts` has a confusing call logic where it keeps creating new Observable to subscribe to in order to paginate exhaustively
  - scattered logic - `src/requestLooper.ts` attempted to generalize the logic by accepting 4 callbacks; it creates a fragmented logic between caller and requestLooper, hence, making the logic flow harder to follow
- this PR attempts to simplify the logic by:
  1. method `paginateQuery` paginate in a more tradition `do...while...` loop to handle asynchronous iteration; this is achievable after turning the `super.query(...)` into `Promise`; it is ok to do so since the SiteWise Query is not streaming and not necessary to apply the Observable pattern;
  1. streamlining the pagination logic under `src/SiteWiseQueryPaginator.ts` (see method `paginateQuery`); append frame data and getting next queries in the same method
- added unit testings for pagination - after the simplification, it allows the unit tests to be added easily

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
- `src/getNextQueries.ts` is copied from `src/DataSource.ts`

### CODE REVIEW

**Review by kevinwcyu — APPROVED** [review]
LGTM. 

Just need to fix the `yarn spellcheck` by fixing the typo `Pagainating` in `src/SiteWiseQueryPaginator.test.ts`. Also `ddced` in the `nextToken` is being picked up as an unknown word by the spellcheck. You can replace that substring in the token with some random characters to make it pass the spellcheck.

**Comment by chejimmy** [comment]
> LGTM.
> 
> Just need to fix the `yarn spellcheck` by fixing the typo `Pagainating` in `src/SiteWiseQueryPaginator.test.ts`. Also `ddced` in the `nextToken` is being picked up as an unknown word by the spellcheck. You can replace that substring in the token with some random characters to make it pass the spellcheck.

🙏 TY for the review; updated for spellcheck 👍

File: cspell.config.json [hunk 1]
```diff
@@ -62,6 +62,8 @@
     "errgroup",
     "ectx",
     "reqs",
-    "struct"
+    "struct",
+    "Pagainating",
+    "pagainating"
```
Comment (kevinwcyu) [inline cspell.config.json:67]: Could you actually remove these from the cspell exclusions, the correct spelling is `Paginating` without the extra "a" after the "g"

Comment (chejimmy) [reply cspell.config.json:67]: oh my bad, I didn't know how to spell
Lemme fix that 👍

Comment (chejimmy) [reply cspell.config.json]: All fixed 👍

**Review by kevinwcyu — APPROVED** [review]

**Pull request merged**